aws/cloudformation/components/www_redirect.yml.erb

```diff
@@ -3,7 +3,24 @@
     # As CDO properties can vary between workstations, using them here is generally unsafe
     # Since this line is restricted to only run in production, we are comfortable using it.
     redirected_domains << "www.#{CDO.pegasus_hostname}" if rack_env?(:production)
+    
+    www_redirect_hosted_zone = Aws::Route53::Client.new.list_hosted_zones_by_name(dns_name: domain).hosted_zones.first
+    raise "Could not find #{domain} in hosted zones" unless www_redirect_hosted_zone.name.delete_suffix('.') == domain
+    www_redirect_hosted_zone_id = www_redirect_hosted_zone.id.delete_prefix("/hostedzone/")
   -%>
+
+  <%=app%>DNS:
+    Type: AWS::Route53::RecordSetGroup
+    Properties:
+      HostedZoneId: <%=www_redirect_hosted_zone_id%>%
+      RecordSets:
+<% redirected_domains.each do |record_set_name| -%>
+        - Name: <%=record_set_name%>
+          Type: A
+          AliasTarget:
+            DNSName: !GetAtt [<%=app%>RedirectDistribution, DomainName]
+            HostedZoneId: Z2FDTNDATAQYW2 # static ID for cloudfront aliases
```

Comment:
just for my own edification: what precisely is being redirected, and why do we want a static ID?

Reply:
`Z2FDTNDATAQYW2` is the Hosted Zone ID for the internal hosted zone within AWS that CloudFront uses to publish domain names for CloudFront Distributions. It's OK to hard code this.

Reply:
To answer your broader question ... `www.[some Pegasus or Dashboard environment]` is being redirected to `[some Pegasus or Dashboard environment]` so that we don't need logic buried in our Ruby web application servers to understand how the subdomain `www.` should be served.

Reply:
Yeah, that's an annoying magic string that AWS has. This DNS record is a change from what's actually implemented. Instead of pointing directly to the Load Balancer, as we do now, this makes it point to the redirect distribution, which is configured to use a lambda (in this file) to 302 redirect to `code.org/...`

I intend to manually test this dns change, but codifying it would be good.

Reply:
I tested 'importing' a recordset into a stack. Real "Imports" aren't supported for route53, so I just created a stack that attempted to create an already-existing record. It failed with the following:

```
[Tried to create resource record set [name='darin.code.org.', type='A'] but it already exists]
```

This means the only option is leaving it separate, or deleting and recreating.